airflow/jobs/backfill_job_runner.py

```diff
@@ -730,25 +732,26 @@ def _per_task_process(key, ti: TaskInstance, session):
                 ti_status.deadlocked.update(ti_status.to_run.values())
                 ti_status.to_run.clear()
 
```

Comment:
I wonder if there is a way to combine these two "for each executor" loops rather than looping twice.  Granted, self.job.executors is going to be small list so the loops aren't huge, but that may be a slight efficiency improvement.

Reply:
Yeah it should be pretty cheap. The lists are small and the loops aren't nested so it's not an N^2 situation it's just 2N (which just reduced to O(N) in the end).